src/core.js

```diff
@@ -533,6 +527,9 @@ ChartInternal.prototype.redraw = function(options, transitions) {
     durationForExit = withTransitionForExit ? duration : 0;
     durationForAxis = withTransitionForAxis ? duration : 0;
 
+    // reset caches when we are re-drawing the chart
+    $$.resetCache();
```

Comment:
I guess this resetting of cache might be inconvenient for some future use cases. How about resetting only some group of caches by the key?

```js
$$.resetCache('MaxTickWidth') // => reset cache which has the key /^\$MaxTickWidth/
```

Reply:
I though about this when I tried to use this method for `legendItemTextBox` too where reseting this cache here would be unnecessary.

I think there will be some kind of tradeoff between ease of use (aka: Forget to reset a cache here & there) & performance.

I think there are 2 families of cached values:

- The ones that are impacted by an update of data (added, removed targets, etc.)
- The ones that are impacted by an update of sizing (resize, update of CSS....)